docs/CONTRIBUTING.md

```diff
@@ -1,11 +1,327 @@
+    dnf -y builddep ostree  && \
+    dnf -y install make redhat-rpm-config && \
+    dnf clean all
+
+# clone your ostree repo
+RUN git clone https://github.com/<username>/ostree.git
+
+# pull patch-branch + set upstream
+RUN cd ostree && \
+    git pull origin <branch of your patch> && \
+    git checkout <branch of your patch> && \
+    git remote add upstream https://github.com/ostreedev/ostree.git
+
+WORKDIR /ostree
+```
+**Build the container**
+
+Run `docker build` in the same directory of the `Dockerfile` like so:
+
+```
+$ docker build -t ostree-fedora-test .
+```
+When this build is done, the `-t` option tags the image as `ostree-fedora-test`.  
+
+**Note**: Do not forget the dot **.** at the end of the above command which specifies the location of the Dockerfile.
+
+
+
+You will see `ostree-fedora-test` listed when running `docker images`:
+
+```bash
+REPOSITORY                                 TAG                 IMAGE ID            CREATED             SIZE
+ostree-fedora-test                         latest              817c04cc3656        1 day ago          978MB
+```
+
+**Entering the Container**
+
+To **start** the `ostree-fedora-test` container, run:
+
+```bash
+$ docker run -it --rm --entrypoint /bin/sh --name ostree-testing ostree-fedora-test
+```
+**Note**:
+
+`--rm` option removes the name of the container i.e. `ostree-testing` whenever `exit` is initiated within the container's shell prompt `#`. This is so that the container name `ostree-testing` can be re-used, otherwise remove it manually by running `docker rm <container name>`.
+
+The state of the container will not be saved when the shell prompt exits. Best practice is modify the Dockerfile to modify the image.
+
+**Testing in a Container Workflow**
+
+1. Edit the changes to OSTree on your local machine.
+2. `git add` your changed files, `git commit -m "commit message"` and then `git push origin <name of local branch>`
+3. From within the `ostree-testing` container, make sure your changes are pulled onto the `ostree` repo in the container. Then run the [ostree build commands](#ostree-build-commands):
+
+```
+git submodule update --init && \
+env NOCONFIGURE=1 ./autogen.sh && \
+./configure && \
+make && \
+make install
+```
+4. `make install` will install OSTree in the default location of the guest OS.
+
+### [Testing in a Virtual Machine](#testing-in-a-virtual-machine)
+
+See this guide on creating an [Atomic Host Vagrant VM](https://gist.github.com/Bubblemelon/bd9f3cf429d9e04be52f9987baed53c8).
+
+To test a build on the Atomic Host Vagrant VM, simply `scp -r <ostree binary location> vagrant@<ip-address>:~`. You can find the IP address of a Vagrant VM by running `vagrant ssh-config` in the same directory as the `Vagrantfile`.
+
+Run `vagrant ssh` to login to the VM. To override the `Read-only file system` warning, run `ostree admin unlock`. Now replace the original OSTree binary with the new one by running `sudo mv ~/ostree /usr/bin/ostree`.
```

Comment:
These bits I think are among the most interesting part of the guide; a lot of the earlier instructions are generic for building any autotools software, and some are generic to C/C++/software in general.

This bit is where we're talking about something that's specific to targeting an ostree-managed system.

On that topic, one thing you *absoutely* need to do is sync *all* of the artifacts from the build - most notably including the `/usr/lib64/libostree-1.so.1.0.0` shared library.

There are two paths to doing that: basically `rsync -rlv ~/ostree-install/usr/ /usr/`, or building an RPM.  Both have advantages/disadvantages; I personally tend to use the rsync path more.

(This is where if we want to get really opinionated a `Vagrantfile` probably makes the most sense?)

Reply:
An even more opinionated path here is for us to just defer to the "Fedora CoreOS SDK".